hallux/targets/gitlab_suggestion.py

```diff
@@ -0,0 +1,166 @@
+# Copyright: Hallux team, 2023
+
+from __future__ import annotations
+
+import os
+from typing import Final
+
+import requests
+
+from ..logger import logger
+from ..proposals.diff_proposal import DiffProposal
+from .filesystem import FilesystemTarget
+from .github_suggestion import GithubSuggestion
+
+
+# Saves Issue Fixes as Github proposals
+class GitlabSuggestion(FilesystemTarget):
+    def __init__(self, mr_url: str | None):
```

Comment:
Refactor this function to reduce its Cognitive Complexity from 16 to the 15 allowed.
```suggestion
    def __init__(self, mr_url: str | None):
```